migration/idempotent/001-base.sql

```diff
@@ -1,16 +1,76 @@
 --NOTES
 --This code assumes that table names can only be 63 chars long
 
+CREATE OR REPLACE VIEW _prom_catalog.initial_default AS
+SELECT *
+FROM
+(
+    VALUES
+    ('chunk_interval'           , (INTERVAL '8 hours')::text),
+    ('retention_period'         , (90 * INTERVAL '1 day')::text),
+    ('metric_compression'       , (exists(select 1 from pg_catalog.pg_proc where proname = 'compress_chunk')::text)),
+    ('trace_retention_period'   , (30 * INTERVAL '1 days')::text),
+    ('ha_lease_timeout'         , '1m'),
+    ('ha_lease_refresh'         , '10s')
+) d(key, value)
+;
+GRANT SELECT ON _prom_catalog.initial_default TO prom_reader;
+
+CREATE OR REPLACE FUNCTION _prom_catalog.get_default_value(_key text)
+    RETURNS TEXT
+    SET search_path = pg_catalog
+AS $func$
+    -- if there is a user-supplied default value, take it
+    -- otherwise take the initial default value
+    SELECT coalesce(d.value, dd.value)
+    FROM _prom_catalog.initial_default dd
+    LEFT OUTER JOIN _prom_catalog.default d ON (dd.key = d.key)
+    WHERE dd.key = _key;
+$func$
+LANGUAGE SQL STABLE PARALLEL SAFE;
+GRANT EXECUTE ON FUNCTION _prom_catalog.get_default_value(text) TO prom_reader;
+
+CREATE OR REPLACE FUNCTION _prom_catalog.set_default_value(_key text, _value text)
+    RETURNS VOID
+    SET search_path = pg_catalog
+AS $func$
+    INSERT INTO _prom_catalog.default (key, value)
+    VALUES (_key, _value)
+    ON CONFLICT (key) DO
+    UPDATE SET value = excluded.value
+    ;
+$func$
+LANGUAGE SQL VOLATILE;
+GRANT EXECUTE ON FUNCTION _prom_catalog.set_default_value(text, text) TO prom_admin;
+
+CREATE OR REPLACE FUNCTION _prom_catalog.is_restore_in_progress()
+RETURNS BOOLEAN
+SET search_path = pg_catalog
+AS $func$
+    SELECT coalesce((SELECT setting::boolean from pg_catalog.pg_settings where name = 'timescaledb.restoring'), false)
+$func$
+LANGUAGE sql STABLE;
+GRANT EXECUTE ON FUNCTION _prom_catalog.is_restore_in_progress() TO prom_reader;
+
 CREATE OR REPLACE PROCEDURE _prom_catalog.execute_everywhere(command_key text, command TEXT, transactional BOOLEAN = true)
     SET search_path = pg_catalog
 AS $func$
+DECLARE
+    _is_restore_in_progress boolean = false;
 BEGIN
     IF command_key IS NOT NULL THEN
        INSERT INTO _prom_catalog.remote_commands(key, command, transactional) VALUES(command_key, command, transactional)
        ON CONFLICT (key) DO UPDATE SET command = excluded.command, transactional = excluded.transactional;
     END IF;
 
     EXECUTE command;
+
+    -- do not call distributed_exec if we are in the middle of restoring from backup
+    _is_restore_in_progress = coalesce((SELECT setting::boolean from pg_catalog.pg_settings where name = 'timescaledb.restoring'), false);
```

Comment:
can't we use _prom_catalog.is_restore_in_progress() here?

Reply:
yep. missed this in refactoring. thanks
resolved in [eda2cee](https://github.com/timescale/promscale_extension/pull/185/commits/eda2ceeea4e1d309986c3034a200707a9f5d661d)